src/jdk.jfr/share/classes/jdk/jfr/internal/util/TimespanRateOrPeriod.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2025 SAP SE. All rights reserved.
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.jfr.internal.util;
+
+import jdk.jfr.internal.settings.CPUThrottleSetting;
+
+/**
+ * A rate or fixed period, see {@link jdk.jfr.internal.Rate}
+ */
+public record TimespanRateOrPeriod(double rate, long periodNanos, boolean isRate) {
+
+    public static final TimespanRateOrPeriod OFF = new TimespanRateOrPeriod(0, 0, false);
+
+    public static TimespanRateOrPeriod of(String text) {
+        if (text.equals("off")) {
+            return OFF;
+        }
+        boolean isPeriod = !text.contains("/");
+        if (isPeriod) {
+            var period = ValueParser.parseTimespanWithInfinity(text, Long.MAX_VALUE);
+            if (period == Long.MAX_VALUE) {
+                return null;
+            }
+            if (period == 0) {
+                return OFF;
+            }
+            return new TimespanRateOrPeriod(0, period, false);
+        }
+        Rate r = Rate.of(text);
+        if (r == null) {
+            return null;
+        }
+        return new TimespanRateOrPeriod(r.perSecond(), 0, true);
+    }
+
+    public static TimespanRateOrPeriod max(TimespanRateOrPeriod a, TimespanRateOrPeriod b) {
```

Comment:
`max` is not the best name for it, since it can be `min` for period or `max` for rate.
The idea of the method is to select a value with the highest resolution, right? Let's reflect this in the name and/or code comment.